src/client/statenav.c

```diff
@@ -346,10 +350,108 @@ void receiveButtonClick(char* value)
       int surrender = strtol(value + strlen("surrender "), NULL, 10);
       sendOrder(SURRENDERORDER, UINT_MAX, surrender);
    }
+   else if (!strncmp(value, "scoreboard", strlen("scoreboard")))
+   {
+      showScoreboard();
+   }
 
    free(value);
 }
 
+void refreshScoreboardIfNecessary()
+{
+   int scoreboardIsVisible = EM_ASM_INT(
+      {
+         return document.getElementById('scoreboard') != null ? 1 : -1;
```

Comment:
There's a slight bug here!

In C, true/false is implemented as integer zero (false) or integer non-zero (true). -1 is also non-zero (true), which results in the scoreboard appearing when stepping history, even if it was closed before stepping.

Reply:
Ahh, of course! Thanks. Not sure why I wrote -1, even with my (extremely) basic C knowledge :)